agr_api/src/main/java/org/alliancegenome/api/controller/GeneController.java

```diff
@@ -271,46 +273,44 @@ public Response getAllelesPerGeneDownload(String id,
 
 		JsonResultResponse<Allele> alleles = geneService.getAlleles(id, pagination);
 
-		Response.ResponseBuilder responseBuilder = Response.ok(alleleTanslator.getAllRows(alleles.getResults()));
+		Response.ResponseBuilder responseBuilder = Response.ok(alleleTranslator.getAllRows(alleles.getResults()));
 		APIServiceHelper.setDownloadHeader(id, EntityType.GENE, EntityType.ALLELE, responseBuilder);
 		return responseBuilder.build();
 	}
 
 
 	@Override
-	public JsonResultResponse<InteractionGeneJoin> getInteractions(String id, Integer limit, Integer page, String sortBy, String asc,
-																String moleculeType,
-																JoinTypeValue joinType,
+	public JsonResultResponse<GeneGeneticInteractionDocument> getGeneticInteractions(String id, Integer limit, Integer page, String sortBy, String asc,
 																String interactorGeneSymbol,
 																String interactorSpecies,
-																String interactorMoleculeType,
-																String detectionMethod,
 																String source,
 																String reference,
 																String role,
 																String geneticPerturbation,
-																String interacotorRole,
+																String interactorRole,
 																String interactorGeneticPerturbation,
 																String phenotypes,
 																String interactionType,
 																@Context UriInfo info) {
 		long startTime = System.currentTimeMillis();
+		
 		Pagination pagination = new Pagination(page, limit, sortBy, asc, new InteractionColumnFieldMapping());
-		pagination.addFieldFilter(FieldFilter.MOLECULE_TYPE, moleculeType);
-		pagination.addFieldFilter(FieldFilter.JOIN_TYPE, joinType.getName());
-		pagination.addFieldFilter(FieldFilter.INTERACTOR_GENE_SYMBOL, interactorGeneSymbol);
-		pagination.addFieldFilter(FieldFilter.INTERACTOR_SPECIES, interactorSpecies);
-		pagination.addFieldFilter(FieldFilter.INTERACTOR_MOLECULE_TYPE, interactorMoleculeType);
-		pagination.addFieldFilter(FieldFilter.DETECTION_METHOD, detectionMethod);
-		pagination.addFieldFilter(FieldFilter.SOURCE, source);
-		pagination.addFieldFilter(FieldFilter.INTERACTOR_REFERENCE, reference);
-		//for genetic interaction
-		pagination.addFieldFilter(FieldFilter.ROLE, role);
-		pagination.addFieldFilter(FieldFilter.GENETIC_PERTURBATION, geneticPerturbation);
-		pagination.addFieldFilter(FieldFilter.INTERACTOR_ROLE, interacotorRole);
-		pagination.addFieldFilter(FieldFilter.INTERACTOR_GENETIC_PERTURBATION, interactorGeneticPerturbation);
-		pagination.addFieldFilter(FieldFilter.PHENOTYPES, phenotypes);
-		pagination.addFieldFilter(FieldFilter.INTERACTION_TYPE, interactionType);
+		pagination.addFilterOption("geneGeneticInteraction.geneGeneAssociationObject.geneSymbol.displayText", interactorGeneSymbol);
```

Comment:
One thing to consider is that the UI back in the day wanted to have these prefixed with "filter." and I think there was a translation between filter name `FieldFilter.NAME` and the actual field name that is on the object. I don't particularly care but I would suggest just take a look at the endpoint in the swagger interface to see if it makes sense.